packages/suite/src/components/firmware/ReconnectDevicePrompt.tsx

```diff
@@ -67,9 +67,10 @@ export const ReconnectDevicePrompt = ({ onClose, onSuccess }: ReconnectDevicePro
     const isWebUsbTransport = useSelector(selectHasTransportOfType('WebUsbTransport'));
     const { showManualReconnectPrompt, status, uiEvent } = useFirmwareInstallation();
     const { device } = useDevice();
+    const eventDevice = uiEvent && 'device' in uiEvent.payload ? uiEvent.payload.device : undefined;
```

Comment:
I'd prefer type narrowing to type which is known to have `payload.device` rather than blindly checking if it exists.
```suggestion
    const eventDevice = uiEvent && uiEvent.type === DEVICE.BUTTON ? uiEvent.payload.device : undefined;
```